sql/src/main/java/org/apache/druid/sql/calcite/rule/ReverseLookupRule.java

```diff
@@ -559,8 +572,16 @@ private RexNode makeMatchCondition(
         } else {
           return SearchOperatorConversion.makeIn(
               reverseLookupKey.arg,
-              stringsToRexNodes(reversedMatchValues, rexBuilder),
+              reversedMatchValues,
+              rexBuilder.getTypeFactory()
+                        .createTypeWithNullability(
+                            rexBuilder.getTypeFactory().createSqlType(SqlTypeName.VARCHAR),
+                            true
+                        ),
               reverseLookupKey.negate,
+
+              // Use regular equals, or SCALAR_IN_ARRAY, depending on inFunctionThreshold.
+              reversedMatchValues.size() >= plannerContext.queryContext().getInFunctionThreshold(),
```

Comment:
I wonder if it would look simpler to pass `plannerContext` instead or `inFunctionThreshold` - and let this logic live inside `makeIn`

Reply:
Ah, it's like this because different usages of this method use different thresholds. Sometimes it's the `inFunctionThreshold`, sometimes it's the `inFunctionExprThreshold`.